packages/react-components/react-persona/src/components/Persona/usePersonaStyles.ts

```diff
@@ -1,33 +1,232 @@
+      gridStyles.secondaryText,
+      optionalTextClassName,
+      state.secondaryText.className,
+    );
+  }
+
+  if (state.tertiaryText) {
+    state.tertiaryText.className = mergeClasses(
+      personaClassNames.tertiaryText,
+      gridStyles.tertiaryText,
+      optionalTextClassName,
+      state.tertiaryText.className,
+    );
+  }
+
+  if (state.quaternaryText) {
+    state.quaternaryText.className = mergeClasses(
+      personaClassNames.quaternaryText,
+      gridStyles.quaternaryText,
+      optionalTextClassName,
+      state.quaternaryText.className,
+    );
+  }
 
   return state;
 };
+
+const useTextStyles = makeStyles({
+  base: {
+    display: 'block',
+  },
+  primaryText: {
+    color: tokens.colorNeutralForeground1,
+  },
+  optionalText: {
+    color: tokens.colorNeutralForeground2,
+  },
+
+  body1: typographyStyles.body1,
+  caption1: typographyStyles.caption1,
+  subtitle2: typographyStyles.subtitle2,
+  subtitle1: typographyStyles.subtitle1,
+  title2: typographyStyles.title2,
+  title3: typographyStyles.title3,
+});
+
+const useTextClassNames = (
+  state: PersonaState,
+): {
+  primaryTextClassName: string;
+  optionalTextClassName: string;
+} => {
+  const { sizing, numTextLines } = state;
+  const textStyles = useTextStyles();
+  const primaryTextClassNames: string[] = [textStyles.base, textStyles.primaryText];
+  const optionalTextClassNames: string[] = [textStyles.base, textStyles.optionalText];
+
+  if (sizing === 'fixed') {
+    if (state.avatar.size) {
+      const { size } = state.avatar;
+
+      if (size < 40) {
+        primaryTextClassNames.push(textStyles.body1);
+        optionalTextClassNames.push(textStyles.caption1);
+      } else if (size < 64) {
+        primaryTextClassNames.push(textStyles.subtitle2);
+        optionalTextClassNames.push(textStyles.body1);
+      } else if (size < 96) {
+        primaryTextClassNames.push(textStyles.subtitle1);
+        optionalTextClassNames.push(textStyles.caption1);
+      } else {
+        if (numTextLines > 1) {
+          primaryTextClassNames.push(textStyles.title3);
+        } else {
+          primaryTextClassNames.push(textStyles.title2);
+        }
+        optionalTextClassNames.push(textStyles.body1);
+      }
+    } else if (state.presence) {
+      const { size } = state.presence;
+
+      if (size === 'extra-small' || size === 'tiny') {
+        if (numTextLines > 1) {
+          primaryTextClassNames.push(textStyles.body1);
+        } else {
+          primaryTextClassNames.push(textStyles.caption1);
+        }
+        optionalTextClassNames.push(textStyles.caption1);
+      } else {
+        primaryTextClassNames.push(textStyles.body1);
+        optionalTextClassNames.push(textStyles.caption1);
+      }
+    }
+  } else {
+    if (numTextLines > 1) {
+      primaryTextClassNames.push(textStyles.body1);
+    } else {
+      primaryTextClassNames.push(textStyles.caption1);
+    }
+    optionalTextClassNames.push(textStyles.caption1);
+  }
```

Comment:
By pushing these classes to an array, it seems to imply that they could get more than one size class, which I'm assuming isn't the case? (It's a little hard to follow all of the branches here).

Could these just be variables you assign like `primaryTextSizeClass = textStyles.body1`, so that it's clear there will be only one? Then the return value of this function could be:
```ts
  return {
    primaryTextClassName: mergeClasses(textStyles.base, textStyles.primaryText, primaryTextSizeClass),
    optionalTextClassName: mergeClasses(textStyles.base, textStyles.optionalText, optionalTextSizeClass),
  };
```